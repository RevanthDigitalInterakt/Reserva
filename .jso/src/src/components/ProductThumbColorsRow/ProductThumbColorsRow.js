  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.default = undefined;
  var _react = _interopRequireWildcard(_$$_REQUIRE(_dependencyMap[0]));
  var _reactNative = _$$_REQUIRE(_dependencyMap[1]);
  function _getRequireWildcardCache(e) { if ("function" != typeof WeakMap) return null; var r = new WeakMap(), t = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(e) { return e ? t : r; })(e); }
  function _interopRequireWildcard(e, r) { if (!r && e && e.__esModule) return e; if (null === e || "object" != typeof e && "function" != typeof e) return { default: e }; var t = _getRequireWildcardCache(r); if (t && t.has(e)) return t.get(e); var n = { __proto__: null }, a = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var u in e) if ("default" !== u && {}.hasOwnProperty.call(e, u)) { var i = a ? Object.getOwnPropertyDescriptor(e, u) : null; i && (i.get || i.set) ? Object.defineProperty(n, u, i) : n[u] = e[u]; } return n.default = e, t && t.set(e, n), n; }
  function ProductThumbColorsRow(_ref) {
    var identifier = _ref.identifier,
      colors = _ref.colors,
      _ref$limit = _ref.limit,
      limit = _ref$limit === undefined ? 4 : _ref$limit;
    var items = (0, _react.useMemo)(function () {
      return Array.from(new Set(colors));
    }, [colors]);
    if (!items.length) return null;
    return /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[2]).jsxs)(_reactNative.View, {
      style: _$$_REQUIRE(_dependencyMap[3]).styles.wrapper,
      children: [items.map(function (item, i) {
        return i < limit && /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[2]).jsx)(_reactNative.View, {
          style: [_$$_REQUIRE(_dependencyMap[3]).styles.circleContainer, {
            backgroundColor: item
          }],
          children: /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[2]).jsx)(_reactNative.View, {
            style: [_$$_REQUIRE(_dependencyMap[3]).styles.circle, {
              backgroundColor: item
            }]
          })
        }, `colorthumb-${identifier}-${item}`);
      }), items.length > limit && /*#__PURE__*/(0, _$$_REQUIRE(_dependencyMap[2]).jsx)(_reactNative.Text, {
        style: _$$_REQUIRE(_dependencyMap[3]).styles.text,
        allowFontScaling: false,
        children: `+${items.length - limit}`
      })]
    });
  }
  var _default = exports.default = ProductThumbColorsRow;
