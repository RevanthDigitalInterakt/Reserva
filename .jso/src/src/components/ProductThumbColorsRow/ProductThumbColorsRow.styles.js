  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.styles = undefined;
  var _reactNative = _$$_REQUIRE(_dependencyMap[0]);
  var styles = exports.styles = _reactNative.StyleSheet.create({
    wrapper: {
      height: 25,
      paddingTop: 10,
      marginBottom: -5,
      flexDirection: 'row'
    },
    circleContainer: {
      width: 15,
      height: 15,
      borderRadius: 100,
      alignItems: 'center',
      justifyContent: 'center',
      marginRight: 8
    },
    circle: {
      width: 12,
      height: 12,
      borderRadius: 100,
      borderWidth: 1.5,
      borderColor: '#ffffff'
    },
    text: {
      color: '#8A8C8E',
      fontSize: 11,
      fontWeight: '400'
    }
  });
