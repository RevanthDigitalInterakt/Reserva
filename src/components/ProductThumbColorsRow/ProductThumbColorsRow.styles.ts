import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  wrapper: {
    height: 25,
    paddingTop: 10,
    marginBottom: -5,
    flexDirection: 'row',
  },

  circleContainer: {
    width: 15,
    height: 15,
    borderRadius: 100,
    alignItems: 'center',
    justifyContent: 'center',
    marginRight: 8,
  },

  circle: {
    width: 12,
    height: 12,
    borderRadius: 100,
    borderWidth: 1.5,
    borderColor: '#ffffff',
  },

  text: {
    color: '#8A8C8E',
    fontSize: 11,
    fontWeight: '400',
  },
});
