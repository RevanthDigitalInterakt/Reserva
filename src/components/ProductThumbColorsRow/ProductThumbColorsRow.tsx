import React, { useMemo } from 'react';
import { Text, View } from 'react-native';
import { styles } from './ProductThumbColorsRow.styles';

interface IProductThumbColors {
  identifier: string;
  colors: string[];
  limit?: number;
}

function ProductThumbColorsRow({ identifier, colors, limit = 4 }: IProductThumbColors) {
  const items = useMemo(() => Array.from(new Set(colors)), [colors]);

  if (!items.length) return null;

  return (
    <View style={styles.wrapper}>
      {items.map((item, i) => (i < limit) && (
        <View
          key={`colorthumb-${identifier}-${item}`}
          style={[styles.circleContainer, { backgroundColor: item }]}
        >
          <View style={[styles.circle, { backgroundColor: item }]} />
        </View>
      ))}

      {(items.length > limit) && (
        <Text style={styles.text} allowFontScaling={false}>{`+${items.length - limit}`}</Text>
      )}
    </View>
  );
}

export default ProductThumbColorsRow;
