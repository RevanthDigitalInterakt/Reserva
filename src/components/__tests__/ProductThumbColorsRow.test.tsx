import React from 'react';
import {
  render,
} from '@testing-library/react-native';
import ProductThumbColorsRow from '../ProductThumbColorsRow/ProductThumbColorsRow';

describe('ProductThumbColorsRow', () => {
  it('should match with the snapshot', async () => {
    const Component = (
      <ProductThumbColorsRow identifier="test" colors={['#fff', '#000']} />
    );

    const { toJSON } = render(Component);

    expect(toJSON()).toMatchSnapshot();
  });
});
