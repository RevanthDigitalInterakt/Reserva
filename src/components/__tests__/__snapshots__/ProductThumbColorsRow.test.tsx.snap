// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductThumbColorsRow should match with the snapshot 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "height": 25,
      "marginBottom": -5,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 100,
          "height": 15,
          "justifyContent": "center",
          "marginRight": 8,
          "width": 15,
        },
        Object {
          "backgroundColor": "#fff",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderColor": "#ffffff",
            "borderRadius": 100,
            "borderWidth": 1.5,
            "height": 12,
            "width": 12,
          },
          Object {
            "backgroundColor": "#fff",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 100,
          "height": 15,
          "justifyContent": "center",
          "marginRight": 8,
          "width": 15,
        },
        Object {
          "backgroundColor": "#000",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderColor": "#ffffff",
            "borderRadius": 100,
            "borderWidth": 1.5,
            "height": 12,
            "width": 12,
          },
          Object {
            "backgroundColor": "#000",
          },
        ]
      }
    />
  </View>
</View>
`;
